TowerDefense/TowerDefense.cs

```diff
@@ -49,7 +49,9 @@ protected override void LoadContent()
 
         protected override void Update(GameTime gameTime)
         {
-            mouse.Update(gameEngine, gameMap);
+            mouse.Update(gameTime, gameEngine, gameMap);
+            gameMap.Update(gameTime);
+            gameHUD.Update(gameTime);
```

Comment:
It's weird to have to call update and handle mouse, handling the mouse sounds like something that should be in update, but some of these didn't even have an update until just now, seems like another area I need to refactor.